tests/test_app_integration.py

```diff
@@ -0,0 +1,187 @@
+
+    return MockResponse(None, 404)
+
+def mock_post_requests(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    if args[0] == 'https://api.trello.com/1/cards/5efcab153d4408623d3a0481':
+        return MockResponse([{'id': '5efcab153d4408623d3a0484', 'name': 'Test task4', 'desc': 'Test in in-progress', 'due': None, 'dateLastActivity': '2020-07-02T12:59:12.694Z'}], 200)
+
+    return MockResponse(None, 404)
+
+def mock_delete_requests(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    if args[0] == 'https://api.trello.com/1/cards/5efcab153d4408623d3a0481':
+        return MockResponse([{'id': '5efcab153d4408623d3a0484', 'name': 'Test task4', 'desc': 'Test in in-progress', 'due': None, 'dateLastActivity': '2020-07-02T12:59:12.694Z'}], 200)
+
+    return MockResponse(None, 404)
+
+
+@mock.patch('requests.get', side_effect=mock_get_requests)
+def test_default_endpoint(mock_get_requests, client):
+    #given
+
+    # when
+    response = client.get("/")
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 200
+    assert "<title>To-Do App</title>" in response_html
+    assert "<h2>Items</h2>" in response_html
+    assert "Test task1" in response_html
+    assert "Test task2" in response_html
+    assert "Test task3" in response_html
+    assert "Test task4" in response_html
+    assert "Test task5" in response_html
+    assert "Test task6" in response_html
+    assert "Test task7" in response_html
+    assert "Test task8" in response_html
+    assert "Test task9" in response_html
+
+@mock.patch('requests.get', side_effect=mock_get_requests)
+@mock.patch('requests.put', side_effect=mock_put_requests)
+def test_update_endpoint(mock_get_requests, mock_put_requests, client):
+    #given
+
+    # when
+    response = client.post("/update/5efcab153d4408623d3a0481/InProgress", follow_redirects=False)
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 302
+    assert "<h1>Redirecting...</h1>" in response_html
+
+def test_add_get_endpoint(client):
+    #given
+
+    # when
+    response = client.get("/add", )
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 200
+    assert "<title>To-Do App</title>" in response_html
+    assert "<h2>Add New Items</h2>" in response_html
+
+@mock.patch('requests.post', side_effect=mock_post_requests)
+def test_add_post_endpoint(mock_post_requests, client):
```

Comment:
Everything seems good except I need to add mock_get_requests to this test to get it to pass